[9.2] Adding a button to the detailed view where the webdav url can be copi…

DeepDiver1975 opened the pull request:
…ed to the clipboard

![bildschirmfoto von 2016-08-10 14-59-34](https://cloud.githubusercontent.com/assets/1005065/17554477/48d1a034-5f0b-11e6-95a5-f9a8f8b5ab3b.png)

The webdav url to a folder can now easily be copied to the clipboard using this new button on the detailed view.

This is valuable for users who want to mount a specific folder in e.g. their windows explorer, nautilus file manager and so on ....

DeepDiver1975 committed:
Adding a button to the detailed view where the webdav url can be copied to the clipboard

cdamken commented:
Thanks @DeepDiver1975 !

PVince81 commented:
- [ ] add some padding !
- [ ] add tooltip, as a user discovering this I have no idea what it is about
- [ ] optionally: change the "Copied" tooltip to "Copied the stuffz" (replace stuffz with whatever it is I copied)

Having these two buttons so close here doesn't feel comfortable and rather confusing. I wonder if we should put them both under the link part, basically it would expand with two fields instead of one. Might look ugly though.

PVince81 commented on apps/files/js/mainfileinfodetailview.js:
```diff
@@ -18,6 +18,7 @@
 				'<span class="icon icon-public"></span>' +
 				'<span class="hidden-visually">{{permalinkTitle}}</span>' +
 			'</a>' +
+			'<a class="{{#unless showWebDavCopyButton}}hidden-visually{{/unless}} clipboardButton icon icon-external" data-clipboard-text="{{webDavUrl}}"></a>' +
```
it should be "hidden", not "hidden-visually".

"hidden-visually" is used for screen readers so they can still read an icon's label even when we don't want to show it visually.

DeepDiver1975 committed:
- add some padding to the button
- use hidden instead of hidden-visually
- add tooltip
- use notifications on copy of the dav url- tooltip is used for the description of the button

PVince81 commented:
I think it would be best to have both links in the dropdown from the permalink to avoid cluttering the UI.

PVince81 commented:
@DeepDiver1975 update or close ?

conflicts detected. also see comments above

PVince81 commented:
considering that most users use the desktop client, I don't see much value in exposing the current Webdav path in such an exposed location. The Webdav path is rather something internal to ownCloud and would qualify as "advanced usage" so should maybe be put somewhere else.

DeepDiver1975 commented:
> considering that most users use the desktop client, I don't see much value in exposing the current Webdav path in such an exposed location.

the use case behind this is that a user can easily copy the web dav path to mount the folder in the operating system's file manager.

seems to be a legitimate use case - maybe? @pmaier1 

originally requested by @MTRichards

PVince81 commented:
now one idea to keep both could be to do like github used to do and have a "tab button" on top of the link that says "Private link" and "Webdav link". It would show "Private link" by default but you can switch to "Webdav link" if you want the Webdav format of it. It's better than two separate fields or two separate buttons.

Apparently Github now changed the git URL and have it with a link instead to switch to "https" mode:
![github-clone](https://user-images.githubusercontent.com/277525/27866422-8c99e360-6196-11e7-85f9-72c946472a23.png)
